c_src/modsec_nif.c

```diff
@@ -242,47 +208,37 @@ static ERL_NIF_TERM modsec_create_ctx(ErlNifEnv *env, int argc, const ERL_NIF_TE
         {
             return enif_make_badarg(env);
         }
-        msc_rules_add_file(ctx->rules, (const char *)conf_file.data, &modsec_error);
+        msc_rules_add_file(rules, (const char *)conf_file.data, &modsec_error);
```

Comment:
I wonder how reloading the rules or rather "replacing" the rules would work (rule update scenario). Have you though about it?